### ISSUE TITLE
Custom keys

### ISSUE THREAD

**Issue by slipdef** [touchlab/Kermit#163]
Is there a way I can associate custom keys with an error? For example Crashlytics allows to do firebaseCrashlytics.setCustomKey.

**Comment by kpgalligan** [comment]
What version are you using? We don't have a way to do that formally, but if you're using one of the newer 1.0.x versions, you can add something like this:

```kotlin
import co.touchlab.kermit.crashlytics.FIRCrashlytics

fun addCrashlyticsKey(key:String, value:Any){
    val cr = FIRCrashlytics.crashlytics()
    cr.setCustomValue(value, key)
}
```

Our cinterop includes `setCustomValue`. You'd need something similar on Android, although that would be easier (no cinterop involved). Could probably make `addCrashlyticsKey` expect/actual. Maybe pass in a collection rather than singles.

We're doing a lot of rework now, so will think about this.

**Comment by slipdef** [comment]
Makes sense. I was thinking also to inherit from Throwable and add custom keys there, then process it in the custom Crashlytics log writer. Anyways native support in the lib would be the best option.
